api/features/models.py

```diff
@@ -528,54 +529,36 @@ def get_environment_flags_list(
         if feature_name:
             feature_states = feature_states.filter(feature__name__iexact=feature_name)
 
-        # Build up a dictionary in the form {feature: feature_state} and only keep the
-        # latest version for each feature.
+        if additional_filters:
+            feature_states = feature_states.filter(additional_filters)
+
+        # Build up a dictionary in the form
+        # {(feature_id, feature_segment_id, identity_id): feature_state}
+        # and only keep the latest version for each feature.
         feature_states_dict = {}
         for feature_state in feature_states:
-            current_feature_state = feature_states_dict.get(feature_state.feature)
+            key = (
+                feature_state.feature_id,
+                feature_state.feature_segment_id,
+                feature_state.identity_id,
+            )
+            current_feature_state = feature_states_dict.get(key)
             if (
                 not current_feature_state
                 or feature_state.version > current_feature_state.version
             ):
-                feature_states_dict[feature_state.feature] = feature_state
+                feature_states_dict[key] = feature_state
```

Comment:
Note that I did a very basic test on the difference in performance between using a tuple of 3 integers and a Feature object and the tuple was ~100x quicker 👍 (obviously we should have been using the feature id rather than the whole feature object but still, we're gaining functionality while also improving the performance).